### PR TITLE
openshift route support

### LINKED ISSUE #111: openshift route support spike
Openshift 4 does is not creating a route when an ingress resource is created.

We need to figure out what needs to be done in order for one to be created, even if we need to create it manually.

One possible option would be to introduce a new "openshift" ingress backend type.

## Acceptance Criteria

This issue is about finding the best way to:

* Manage ingresses as it is in kubernetes
* Create a openshift route if running on openshift

## Definition of Done

Find the best way to manage both ingress and openshift route in the operator.

### PR DESCRIPTION
**Pull request by odra**
Fixes #111 

## Proposed Changes

* Adds openshift route support;
* Adds a new `ingress_backend` type for openshift.

## Verification Steps

* Run tests;
* Deploy on openshift

## Additional Notes

Using openshift with local tests won't work since it uses raw kuberetes api-server.

### CODE REVIEW

File: mbox-operator/roles/koji-hub/tasks/main.yml [hunk 1]
```diff
@@ -124,7 +128,23 @@
...
-  when: koji_hub_http_enabled
+  when: koji_hub_ingress_backend != 'openshift'
+
+- block:
+    - name: Apply koji-hub openshift route
+      template:
+        src: openshift-route.yml.j2
+        dest: /tmp/openshift-route.yml
+      vars:
+        koji_hub_route_host: "{{ koji_hub_host }}"
```
Comment (Zlopez) [inline mbox-operator/roles/koji-hub/tasks/main.yml:139]: This var is not used in `openshift-route.yml.j2`.

**Review by Zlopez — APPROVED** [review]
Looks good, just one note

**Comment by odra** [comment]
recheck

**Comment by odra** [comment]
recheck

**Comment by odra** [comment]
Some  unstable networking issue in zuul.. I will merge this and see if we can hook a jenkins instance up here instead of zuul.

**Pull request merged**